### PR TITLE
Dogfood the Kotlin DSL

### PR DESCRIPTION
**Pull request by bamboo**

### CODE REVIEW

File: buildSrc/src/main/kotlin/codegen/GenerateClasspathManifest.kt [hunk 1]
```diff
@@ -34,10 +37,10 @@ open class GenerateClasspathManifest : DefaultTask() {
     var outputDirectory: File? = null
 
     @get:InputFiles
-    val compileOnly = project.configurations.getByName("compileOnly")
+    val compileOnly = project.configurations["compileOnly"]
```
Comment (eskatos) [inline buildSrc/src/main/kotlin/codegen/GenerateClasspathManifest.kt]: :o: DRY with `val compileOnly by project.configurations`

File: buildSrc/src/main/kotlin/codegen/GenerateClasspathManifest.kt [hunk 2]
```diff
@@ -34,10 +37,10 @@ open class GenerateClasspathManifest : DefaultTask() {
...
 
     @get:InputFiles
-    val runtime = project.configurations.getByName("runtime")
+    val runtime = project.configurations["runtime"]
```
Comment (eskatos) [inline buildSrc/src/main/kotlin/codegen/GenerateClasspathManifest.kt]: :o: DRY with `val runtime by project.configurations`

File: buildSrc/src/main/kotlin/codegen/GenerateKotlinDependencyExtensions.kt [hunk 3]
```diff
@@ -67,7 +64,7 @@ val embeddedKotlinVersion = "$embeddedKotlinVersion"
  */
 @Deprecated("Will be removed in 1.0")
 fun RepositoryHandler.gradleScriptKotlin(): ArtifactRepository =
-    maven { it.setUrl("$kotlinDslRepository") }
+    gradlePluginPortal()
```
Comment (eskatos) [inline buildSrc/src/main/kotlin/codegen/GenerateKotlinDependencyExtensions.kt]: ❓Could you explain why this change?

File: buildSrc/src/main/kotlin/plugins/kotlin-dsl-module.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ *  - include all sources
+ */
+
+import accessors.java
+
+
+apply(plugin = "kotlin-library")
+
+// including all sources
+val mainSourceSet = java.sourceSets["main"]
```
Comment (eskatos) [inline buildSrc/src/main/kotlin/plugins/kotlin-dsl-module.gradle.kts]: :o: `val main by java.sourceSets` ?

File: buildSrc/src/main/kotlin/plugins/kotlin-dsl-plugin-bundle.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+
+afterEvaluate {
+
+    pluginBundle {
+        mavenCoordinates.artifactId = base.archivesBaseName
+    }
+
+    kotlinDslPlugins.all {
```
Comment (eskatos) [inline buildSrc/src/main/kotlin/plugins/kotlin-dsl-plugin-bundle.gradle.kts]: :o: this block could go outside of the `afterEvaluate {}` block

File: buildSrc/src/main/kotlin/plugins/public-kotlin-dsl-module.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    }
+}
+
+tasks.getByName("artifactoryPublish") {
+    dependsOn("jar")
+}
+
+// classpath manifest
+val generatedResourcesDir = file("$buildDir/generate-resources/main")
+val generateClasspathManifest = tasks.create<GenerateClasspathManifest>("generateClasspathManifest") {
```
Comment (eskatos) [inline buildSrc/src/main/kotlin/plugins/public-kotlin-dsl-module.gradle.kts]: :o: DRY with `val generateClasspathManifest by tasks.creating(GenerateClasspathManifest::class) { }`

File: buildSrc/src/main/kotlin/plugins/public-kotlin-dsl-module.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    outputDirectory = generatedResourcesDir
+}
+val mainSourceSet = java.sourceSets["main"]
```
Comment (eskatos) [inline buildSrc/src/main/kotlin/plugins/public-kotlin-dsl-module.gradle.kts]: :o: DRY with `val main by java.sourceSets`

File: buildSrc/src/main/kotlin/codegen/GenerateKotlinDependencyExtensions.kt (hunk 3 above)
Comment (bamboo) [reply buildSrc/src/main/kotlin/codegen/GenerateKotlinDependencyExtensions.kt]: `kotlinDslRepository` was always the constant `"https://plugins.gradle.org/m2/"` which got removed in 027141713ee6470d768eda980b36b81d6a5ed449 because unnecessary.

File: build.gradle.kts [hunk 8]
```diff
@@ -176,9 +175,8 @@ idea {
 
 
 // --- Utility functions -----------------------------------------------
-operator fun Regex.contains(s: String) = matches(s)
-
-inline fun <reified T : Task> task(noinline configuration: T.() -> Unit) = tasks.creating(T::class, configuration)
+inline fun <reified T : Task> task(noinline configuration: T.() -> Unit) =
+    tasks.creating(T::class, configuration)
```
Comment (JLLeitschuh) [inline build.gradle.kts]: Would love this if it was just part of the public API.
